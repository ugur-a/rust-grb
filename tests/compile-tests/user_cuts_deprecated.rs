#![deny(deprecated)]

use grb::callback::*;
use grb::prelude::*;

fn callback(w: Where) -> CbResult {
    match w {
        Where::MIPSol(ctx) => {
            if let Where::MIPSol(ctx) = w {
                ctx.add_cut(c!(0 == 0))?;
            }
        }
        Where::MIPNode(ctx) => {
            ctx.add_cut(c!(0 == 0))?;
        }
    }
    Ok(())
}

fn main() {}
