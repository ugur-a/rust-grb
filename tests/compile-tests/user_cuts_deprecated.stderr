error: use of deprecated method `grb::callback::MIPSolCtx::<'a>::add_cut`: This method does nothing, use `MIPNodeCtx::add_cut` instead.
 --> tests/compile-tests/user_cuts_deprecated.rs:9:17
  |
9 |             ctx.add_cut(c!(0 == 0))?;
  |                 ^^^^^^^
  |
note: the lint level is defined here
 --> tests/compile-tests/user_cuts_deprecated.rs:1:9
  |
1 | #![deny(deprecated)]
  |         ^^^^^^^^^^
